tools/topology/development/sof-apl-pcm512x-dual-ssp.m4

```diff
@@ -0,0 +1,201 @@
+#
+# Topology for generic Apollolake UP^2 with pcm512x codec and HDMI.
+#
+
+# if XPROC is not defined, define with default pipe
+ifdef(`DMICPROC', , `define(DMICPROC, eq-iir-volume)')
+ifdef(`DMIC16KPROC', , `define(DMIC16KPROC, eq-iir-volume)')
+
+# if CHANNELS is not defined, define with default 2ch. Note that
+# it can be overrode with DMIC_DAI_CHANNELS, DMIC_PCM_CHANNELS
+# in intel-generic-dmic.m4. Same macros exist for DMIC16K too.
+ifdef(`CHANNELS', , `define(CHANNELS, 2)')
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`ssp.m4')
+include(`hda.m4')
+include(`enumcontrol.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Apollolake DSP configuration
+include(`platform/intel/bxt.m4')
+
+define(`SSP_SCHEDULE_TIME_DOMAIN',
+	ifdef(`CODEC_MASTER', SCHEDULE_TIME_DOMAIN_DMA, SCHEDULE_TIME_DOMAIN_TIMER))
+
+DEBUG_START
+
+#
+# Define the pipelines
+#
+# PCM0 <---> volume <----> SSP5 (pcm512x)
+# PCM5 ----> volume -----> iDisp1
+# PCM6 ----> volume -----> iDisp2
+# PCM7 ----> volume -----> iDisp3
+# PCM4 ----> volume -----> Media Playback 4
+# PCM1 <------------------ DMIC0 (DMIC)
+# PCM2 <------------------ DMIC1 (DMIC16kHz)
+#
+
+dnl PIPELINE_PCM_ADD(pipeline,
+dnl     pipe id, pcm, max channels, format,
+dnl     period, priority, core,
+dnl     pcm_min_rate, pcm_max_rate, pipeline_rate,
+dnl     time_domain, sched_comp)
+
+# Low Latency playback pipeline 1 on PCM 0 using max 2 channels of s32le.
+# Set 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-passthrough-playback-bt.m4,
+	1, 0, 2, s32le,
+	1000, 0, 0,
+	8000, 48000, 48000)
+
+# Low Latency capture pipeline 2 on PCM 0 using max 2 channels of s32le.
+# 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-volume-capture.m4,
+	6, 0, 2, s32le,
+	1000, 0, 0,
+	FSYNC, FSYNC, FSYNC)
```

Comment:
why is playback different from capture for Bluetooth? SSP needs to have the same configuration on playback and capture anyways so not getting the idea at all.

Reply:
yes I will change this.